components/butterfly/src/rumor/dat_file.rs

```diff
@@ -96,6 +96,7 @@ impl DatFileReader {
         let size = OpenOptions::new().create(true)
                                      .read(true)
                                      .write(true)
+                                     .truncate(true)
```

Comment:
This should be `false` which will fix the rumor unit test. It looks like function checks the size of the file using this `create` call. Not entirely sure why this approach was used, but truncating it will obviously wipe it for subsequent reading.